sdk/core/System.ClientModel/src/ModelReaderWriter/ModelJsonConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A generic converter which allows <see cref="JsonSerializer"/> to be able to write and read any models that implement <see cref="IJsonModel{T}"/>.
+/// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+[RequiresUnreferencedCode("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+#endif
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+public class ModelJsonConverter : JsonConverter<IJsonModel<object>>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+{
+    /// <summary>
+    /// Gets the <see cref="ModelReaderWriterOptions"/> used to read and write models.
+    /// </summary>
+    public ModelReaderWriterOptions Options { get; }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/> with a default options of <see cref="ModelReaderWriterOptions.Json"/>.
+    /// </summary>
+    public ModelJsonConverter()
+        : this(ModelReaderWriterOptions.Json) { }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+    public ModelJsonConverter(ModelReaderWriterOptions options)
+    {
+        Options = options;
+    }
+
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return !Attribute.IsDefined(typeToConvert, typeof(JsonConverterAttribute));
+    }
+
+    /// <inheritdoc/>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public override IJsonModel<object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
do we honor the JsonSerializerOptions?

Reply:
No, there are no properties of those options that would apply.

Originally we had a bool property on MRWO that indicated if you wanted additional properties so we did honor the JSO equivalent and passed that through.  After we swapped to using format only to indicate this there is nothing to pull over because there is no 1 to 1 equivalent settings between the two.

Basically if you use this converter you are delegating all logic to MRW.

We punted things like `pretty print` concepts as well from MRWO, but this would be something we would mirror so that the sub tree owned by MRW would look the same as the rest of the json document when written.